tests/test_representative_time.py

```diff
@@ -0,0 +1,130 @@
+        DateTimeDimensionModel(
+            name="Time EST",
+            display_name="Time EST",
+            description="test",
+            class_name="Time",
+            type=DimensionType.TIME,
+            time_type=TimeDimensionType.DATETIME,
+            measurement_type=MeasurementType.TOTAL,
+            str_format="%Y-%m-%d %H:%M:%S",
+            ranges=[
+                TimeRangeModel(
+                    start="2018-01-01 00:00:00",
+                    end="2018-12-31 23:00:00",
+                ),
+            ],
+            frequency=timedelta(hours=1),
+            timezone=TimeZone.EST,
+            time_interval_type=TimeIntervalType.PERIOD_BEGINNING,
+        )
+    )
+
+
+def make_one_weekday_day_and_one_weekend_day_per_month_by_hour_config():
+    return RepresentativePeriodTimeDimensionConfig(
+        RepresentativePeriodTimeDimensionModel(
+            name="one_weekday_day_and_one_weekend_day_per_month_by_hour",
+            display_name="Hourly for representative days",
+            description="test",
+            class_name="Time",
+            type=DimensionType.TIME,
+            time_type=TimeDimensionType.REPRESENTATIVE_PERIOD,
+            measurement_type=MeasurementType.TOTAL,
+            format=RepresentativePeriodFormat.ONE_WEEKDAY_DAY_AND_ONE_WEEKEND_DAY_PER_MONTH_BY_HOUR,
+            ranges=[MonthRangeModel(start=1, end=12)],
+            time_interval_type=TimeIntervalType.PERIOD_ENDING,
+        )
+    )
+
+
+def test_time_consistency(one_weekday_day_and_one_weekend_day_per_month_by_hour_table):
+    df = one_weekday_day_and_one_weekend_day_per_month_by_hour_table
+    config = make_one_weekday_day_and_one_weekend_day_per_month_by_hour_config()
+    time_columns = config.get_load_data_time_columns()
+    assert time_columns == ["month", "is_weekday", "hour"]
+    with pytest.raises(DSGInvalidDataset):
+        config.check_dataset_time_consistency(df.filter("hour != 3"), time_columns)
+    config.check_dataset_time_consistency(df, time_columns)
+
+
+def test_time_mapping(one_weekday_day_and_one_weekend_day_per_month_by_hour_table):
+    df = one_weekday_day_and_one_weekend_day_per_month_by_hour_table
+    # This dataset has only California counties.
+    df = df.withColumn("time_zone", F.lit("PacificStandard"))
+    config = make_one_weekday_day_and_one_weekend_day_per_month_by_hour_config()
+    project_time_config = make_date_time_config()
+    mapped_df = config.convert_dataframe(df, project_time_config)
+    timestamps = mapped_df.select("timestamp").distinct().sort("timestamp").collect()
+    zi = ZoneInfo("EST")
+    est_timestamps = [x.timestamp.astimezone(zi) for x in timestamps]
+    start = datetime(year=2018, month=1, day=1, tzinfo=zi)
+    resolution = timedelta(hours=1)
+    expected_timestamps = [start + i * resolution for i in range(8760)]
+    assert est_timestamps == expected_timestamps
+    assert mapped_df.filter(f"{VALUE_COLUMN} IS NULL").rdd.isEmpty()
```

Comment:
In this test, could we check that the peak time is happening at the expected timestamp?